[MLIR][Affine] Fix a crash with invalid cachesize

SwapnilGhanshyala opened the pull request:
Updated LoopTiling::runOnOperation() to signal pass failure incase the set cachesize is equal to zero.

 Fixes #64979.

CoTinker commented on mlir/lib/Dialect/Affine/Transforms/LoopTiling.cpp:
```diff
@@ -173,6 +173,12 @@ void LoopTiling::getTileSizes(ArrayRef<AffineForOp> band,
 
 void LoopTiling::runOnOperation() {
+  if (cacheSizeInKiB <= 0) {
+    getOperation().emitError(
```
I think use `mlir::emitError` is better, and `cacheSizeInKiB` is uint64_t, it's not less than zero. Maybe you can change it to:
```
if (cacheSizeInKiB == 0) {
  mlir::emitError(mlir::UnknownLoc::get(&Pass::getContext()),
                  "cache size cannot be 0");
}
```
And please add a test. You can refer to #68762.

SwapnilGhanshyala replied:
Thank you for the pointers! Changes made to in the new commit.

Commit:
[MLIR][Affine]Fixed crash with invalid cachesize (Issue #64979)

Updated LoopTiling::runOnOperation() to signal pass failure incase the set cachesize is invalid, i.e., less than or equal to zero.
 #64979 is the reporting issue. Also added a test case "loop-tile-cache-size-invalid.mlir".

CoTinker commented on mlir/test/Dialect/Affine/loop-tile-cache-size-invalid.mlir:
```diff
@@ -0,0 +1,14 @@
+    affine.store %2, %0[%i] : memref<64xf32>
+  }
+  return
+}
```
Maybe you should add a blank line at the end of file.

SwapnilGhanshyala replied:
Sure, addressed in the latest commit!

bondhugula commented:
Cache size zero can be considered a valid input and should not signal pass failure for it. Fixed here: https://github.com/llvm/llvm-project/pull/130526
